src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java

```diff
@@ -131,6 +133,82 @@ public void run() {
     }
 
     /**
+     * Look for running libsyncthing.so processes and kill them.
+     * Try a SIGTERM once, then try again (twice) with SIGKILL.
+     *
+     */
+    public static void killSyncthing() {
+        for (int i = 0; i < 2; i++) {
+            Process ps = null;
+            DataOutputStream psOut = null;
+            try {
+                ps = Runtime.getRuntime().exec("sh");
+                psOut = new DataOutputStream(ps.getOutputStream());
+                psOut.writeBytes("ps | grep libsyncthing.so\n");
+                psOut.writeBytes("exit\n");
+                psOut.flush();
+                ps.waitFor();
+                InputStreamReader isr = new InputStreamReader(ps.getInputStream());
+                BufferedReader br = new BufferedReader(isr);
+                String id;
+                while ((id = br.readLine()) != null) {
+                    killProcessId(id, i > 0);
+                }
+            } catch (IOException | InterruptedException e) {
+                Log.w(TAG_KILL, "Failed list Syncthing processes", e);
+            } finally {
+                try {
+                    if (psOut != null) {
+                        psOut.close();
+                    }
+                } catch (IOException e) {
+                    Log.w(TAG_KILL, "Failed close the psOut stream", e);
+                }
+                if (ps != null) {
+                    ps.destroy();
+                }
+            }
+        }
+    }
+
+    /**
+     * Kill a given process ID
+     *
+     * @param force Whether to use a SIGKILL.
+     */
+    private static void killProcessId(String id, boolean force) {
+        Process kill = null;
+        DataOutputStream killOut = null;
+        try {
+            kill = Runtime.getRuntime().exec("sh");
+            killOut = new DataOutputStream(kill.getOutputStream());
+            if (!force) {
+                killOut.writeBytes("kill " + id + "\n");
+                killOut.writeBytes("sleep 1\n");
+            } else {
+                killOut.writeBytes("sleep 3\n");
+                killOut.writeBytes("kill -9 " + id + "\n");
```

Comment:
Yeah but I don't think we really need the SIGKILL, I've never had trouble to kill a command line program with CTRL-C. Like I said, this could be a problem if a new syncthing instance is started after the call, because it would then be killed.